lib/ansiblelint/text.py

```diff
@@ -0,0 +1,76 @@
+"""Text utils."""
+import re
+import subprocess
+from typing import TYPE_CHECKING, Any, List, Union
+
+from ansiblelint._internal.rules import BaseRule, RuntimeErrorRule
+from ansiblelint.errors import MatchError
+from ansiblelint.rules.AnsibleSyntaxCheckRule import AnsibleSyntaxCheckRule
+
+if TYPE_CHECKING:
+    # https://github.com/PyCQA/pylint/issues/3240
+    # pylint: disable=unsubscriptable-object
+    CompletedProcess = subprocess.CompletedProcess[Any]
+else:
+    CompletedProcess = subprocess.CompletedProcess
+
+_ansible_syntax_check_re = re.compile(
+    r"^ERROR! (?P<title>[^\n]*)\n\nThe error appears to be in "
+    r"'(?P<filename>.*)': line (?P<line>\d+), column (?P<column>\d+)",
+    re.MULTILINE | re.S | re.DOTALL)
+
+
+def strip_ansi_escape(data: Union[str, bytes]) -> str:
+    """Remove all ANSI escapes from string or bytes.
+
+    If bytes is passed instead of string, it will be converted to string
+    using UTF-8.
+    """
+    if isinstance(data, bytes):
+        data = data.decode("utf-8")
+
+    return re.sub(r"\x1b[^m]*m", "", data)
+
+
+def parse_ansible_syntax_check(run: CompletedProcess) -> List[MatchError]:
```

Comment:
Why is a text-parsing function taking process as an input? It should just receive the Ansible output and parse it. Also, I am not sure if this is really a text-related function at all since it returns a list of error messages. I would expect this kind of function to be in some Ansible wrapper module.

Reply:
I am totally for isolating Ansible dependency on the linter but that is not really a case of code depending on Ansible code, is almost only text processing.

But after looking a bit more at how is used, I will make it a method of the runner and keep only the stripping inside text module (others methods will be moved there in the future, once we strip more out of the utils.py kitchensink)